bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,908 @@
+        thing.setDeviceId(normalizedDeviceId);
+        VerisureThing oldObj = verisureThings.get(normalizedDeviceId);
+        logger.trace("Old thing: {}", oldObj);
+        logger.trace("Updated thing: {}", thing);
+        if (oldObj == null || !oldObj.equals(thing)) {
+            thing.setSiteId(installation.getInstallationId());
+            thing.setSiteName(installation.getInstallationName());
+            verisureThings.put(thing.getDeviceId(), thing);
+            logger.trace("Notify listener of thing {}", thing);
+            notifyListeners(thing);
+        }
+    }
+
+    private void updateStatus() {
+        logger.debug("VerisureSession:updateStatus");
+
+        verisureInstallations.forEach((installationId, installation) -> {
+            configureInstallationInstance(installation.getInstallationId());
+            int httpResultCode = setSessionCookieAuthLogin();
+            if (httpResultCode == HttpStatus.OK_200) {
+                updateAlarmStatus(VerisureAlarms.class, installation);
+                updateSmartLockStatus(VerisureSmartLocks.class, installation);
+                updateMiceDetectionStatus(VerisureMiceDetection.class, installation);
+                updateClimateStatus(VerisureClimates.class, installation);
+                updateDoorWindowStatus(VerisureDoorWindows.class, installation);
+                updateUserPresenceStatus(VerisureUserPresences.class, installation);
+                updateSmartPlugStatus(VerisureSmartPlugs.class, installation);
+                updateBroadbandConnectionStatus(VerisureBroadbandConnections.class, installation);
+            } else {
+                logger.warn("Failed to set session cookie and auth login, HTTP result code: {}", httpResultCode);
+            }
+        });
+    }
+
+    private String createOperationJSON(String operation, BigDecimal installationId, String query) {
+        ArrayList<Operation> list = new ArrayList<>();
+        Operation operationJSON = new Operation();
+        Variables variables = new Variables();
+
+        variables.setGiid(installationId.toString());
+        operationJSON.setOperationName(operation);
+        operationJSON.setVariables(variables);
+        operationJSON.setQuery(query);
+        list.add(operationJSON);
+
+        return gson.toJson(list);
+    }
+
+    private synchronized void updateAlarmStatus(Class<? extends VerisureThing> jsonClass,
+            VerisureInstallation installation) {
+        BigDecimal installationId = installation.getInstallationId();
+        String url = START_GRAPHQL;
+        String operation = "ArmState";
+        String query = "query " + operation
+                + "($giid: String!) {\n  installation(giid: $giid) {\n    armState {\n      type\n      statusType\n      date\n      name\n      changedVia\n      allowedForFirstLine\n      allowed\n      errorCodes {\n        value\n        message\n        __typename\n      }\n      __typename\n    }\n    __typename\n  }\n}\n";
```

Comment:
What's with all the extra spaces in this string?

Reply:
That is exactly how the query is sent from the webpage, but totally uneccessary so I've only added one space for readability.